### PR TITLE
Implement vm.ps4dev.trcmem_total

### LINKED ISSUE #486: Implement sysctl vm.ps4dev.trcmem_total

### PR DESCRIPTION
**Pull request by SuchAFuriousDeath**
Should close #486.

### CODE REVIEW

**Review by ultimaweapon — CHANGES_REQUESTED** [review]

File: src/kernel/src/sysctl/mod.rs [hunk 1]
```diff
@@ -846,6 +848,56 @@ static KERN_SMP_CPUS: Oid = Oid {
...
+static VM_PS4DEV_CHILDREN: OidList = OidList {
+    first: Some(&VM_PS4DEV_TRCMEM_TOTAL), // TODO: Use a proper value.
+};
+
+static VM_PS4DEV_TRCMEM_TOTAL: Oid = Oid {
+    parent: &VM_PS4DEV_CHILDREN,
+    link: None, // TODO: Change to a proper value.
+    number: Sysctl::VM_PS4DEV_TRCMEM_TOTAL,
+    kind: Sysctl::CTLFLAG_RD | Sysctl::CTLFLAG_MPSAFE | Sysctl::CTLTYPE_UINT,
+    arg1: None, //TODO: This value on the PS4 is not null.
```
[the remark this answers is not in the document]
Comment (SuchAFuriousDeath) [reply src/kernel/src/sysctl/mod.rs:892]: I assume it's a reference to a static int, but I don't know what the initial value is.

File: src/kernel/src/sysctl/mod.rs [hunk 2]
```diff
@@ -89,6 +89,8 @@ impl Sysctl {
     pub const MACHDEP_TSC_FREQ: i32 = 492;
 
     pub const VM_TOTAL: i32 = 1;
```
[the remark this answers is not in the document]
Comment (SuchAFuriousDeath) [reply src/kernel/src/sysctl/mod.rs]: Done

File: src/kernel/src/sysctl/mod.rs [hunk 3]
```diff
@@ -846,6 +848,56 @@ static KERN_SMP_CPUS: Oid = Oid {
...
+    link: None, // TODO: Change to a proper value.
+    number: Sysctl::VM_PS4DEV_TRCMEM_TOTAL,
+    kind: Sysctl::CTLFLAG_RD | Sysctl::CTLFLAG_MPSAFE | Sysctl::CTLTYPE_UINT,
+    arg1: None, //TODO: This value on the PS4 is not null.
+    arg2: 0,
+    name: "trcmem_total",
+    handler: Some(Sysctl::handle_int),
+    fmt: "IU",
+    descr: "trace memory total",
+    enabled: false,
```
[the remark this answers is not in the document]
Comment (SuchAFuriousDeath) [reply src/kernel/src/sysctl/mod.rs]: Fixed, thanks.

File: src/kernel/src/sysctl/mod.rs [hunk 4]
```diff
@@ -846,6 +848,56 @@ static KERN_SMP_CPUS: Oid = Oid {
     enabled: true,
 };
 
+static VM: Oid = Oid {
+    parent: &CHILDREN,
+    link: Some(&HW),
```
[the remark this answers is not in the document]
Comment (SuchAFuriousDeath) [reply src/kernel/src/sysctl/mod.rs:852]: Should I just add a comment, or fill in the actual in between nodes?

File: src/kernel/src/sysctl/mod.rs [hunk 5]
```diff
@@ -846,6 +848,56 @@ static KERN_SMP_CPUS: Oid = Oid {
...
+    parent: &VM_CHILDREN,
+    link: None, // TODO: Change to a proper value.
+    number: Sysctl::VM_PS4DEV,
+    kind: Sysctl::CTLFLAG_RD | Sysctl::CTLTYPE_NODE,
+    arg1: Some(&VM_PS4DEV_CHILDREN),
+    arg2: 0,
+    name: "ps4dev",
+    handler: None,
+    fmt: "N",
+    descr: "vm parameters for PS4(DevKit only)",
```
[the remark this answers is not in the document]
Comment (SuchAFuriousDeath) [reply src/kernel/src/sysctl/mod.rs:878]: You mean the string? What should it be?

File: src/kernel/src/sysctl/mod.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (ultimaweapon) [reply src/kernel/src/sysctl/mod.rs:852]: Just a TODO comment like the other ones so the people know there are some works required here.

File: src/kernel/src/sysctl/mod.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (ultimaweapon) [reply src/kernel/src/sysctl/mod.rs:878]: `vm parameters for PS4 (DevKit only)`

File: src/kernel/src/sysctl/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ultimaweapon) [reply src/kernel/src/sysctl/mod.rs:891]: It is zero.

File: src/kernel/src/sysctl/mod.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (SuchAFuriousDeath) [reply src/kernel/src/sysctl/mod.rs]: Totally missed that, thanks.

**Comment by SuchAFuriousDeath** [comment]
@ultimaweapon Everything should be fixed now.

File: src/kernel/src/sysctl/mod.rs [hunk 6]
```diff
@@ -846,6 +847,58 @@ static KERN_SMP_CPUS: Oid = Oid {
...
+    arg2: 0,
+    name: "vm",
+    handler: None,
+    fmt: "N",
+    descr: "Virtual memory",
+    enabled: false,
+};
+
+static VM_CHILDREN: OidList = OidList {
+    first: Some(&VM_PS4DEV), // TODO: Use a proper value.
```
Comment (ultimaweapon) [inline src/kernel/src/sysctl/mod.rs:865]: Is not this already correct?

**Review by ultimaweapon — CHANGES_REQUESTED** [review]

File: src/kernel/src/sysctl/mod.rs (hunk 6 above)
Comment (SuchAFuriousDeath) [reply src/kernel/src/sysctl/mod.rs:865]: I suppose it is.

**Comment by SuchAFuriousDeath** [comment]
@ultimaweapon What about now? :D

**Review by ultimaweapon — APPROVED** [review]

**Pull request merged**